gensim/models/keyedvectors.py

```diff
@@ -507,11 +508,12 @@ def most_similar(self, positive=None, negative=None, topn=10, restrict_vocab=Non
 
         Returns
         -------
-        list of (str, float)
-            Sequence of (word, similarity).
+        {list of (str, float), numpy.array}
+            Sequence of (word, similarity) when `topn` is int. When `topn` is None, then
+            similarities for all words are returned.
```

Comment:
I think mentioning the dimensionality of the array is helpful here.

```suggestion
            similarities for all words are returned as a numpy.array.
            The array will have N rows and M columns, where N is ... and M is ....
```

Reply:
Added in 7594c9ec.